bench/macro/Bench/Database/LSMTree/Internal/Lookup.hs

```diff
@@ -450,13 +451,13 @@ benchPrepLookups !bs !ics !hs !keyRng !n
   | n <= 0 = pure ()
   | otherwise = do
       let (!ks, !keyRng') = genLookupBatch keyRng benchmarkGenBatchSize
-      (!_rkixs, !_ioops) <- prepLookups bs ics hs ks
+      (!_rkixs, !_ioops) <- stToIO $ prepLookups bs ics hs ks
       benchPrepLookups bs ics hs keyRng' (n-benchmarkGenBatchSize)
 
 -- | This gives us the combined cost of calculating batches of keys, and
 -- performing disk lookups for each batch.
 benchLookupsInBatches ::
-     (PrimMonad m, MonadAsync m, MonadThrow m)
+     (PrimMonad m, MonadAsync m, MonadThrow m, MonadST m)
```

Comment:
It's unclear that it can use the specialised IO version of `lookupsInBatches` if this itself is not specialised to IO. It might, but it'd rely on inlining within the benchmark itself.

Reply:
Good point. I'm going to fix the type to `IO` for these benchmarkable functions, since we're not using them at any other type anyway